enterprise/models.py

```diff
@@ -272,6 +272,37 @@ def user_email(self):
             return self.user.email
         return None
 
+    @property
+    def entitlements(self):
```

Comment:
@haikuginger Hmm, I thought the opposite - whether it's course level consent or account-level consent, if the learner must pay for the course, I thought we always want to prompt them for consent before they pay.

If GriffCo is using course-level consent only, and users go to use an entitlement that gives them 50% off, they must be prompted to agree to data sharing before they pay (hence before they enroll). Right? We don't want users to pay for a course and then disagree with data sharing, since then they've paid for something they cannot use.

> A user who is logged in, but has not provided consent, is completely ineligible to redeem offers from an EC who requires account-level consent.

Yes, that makes sense.

> In contrast, we should not gate on course-specific consent, because such a consent cannot be provided until after the user attempts to take advantage of the offer.

When you say "cannot" here, you mean "cannot currently due to how we implemented course-level consent", rather than "cannot in principle", right?

I seem to recall that this is a mistake due to us technically tying consent to the enrollment, and is not the desired behavior. I think we talked about changing this and using cookies or some other way to record per-course consent before enrollment takes place.